apps/frontend/src/app/profile/profile.component.ts

```diff
@@ -0,0 +1,168 @@
+      }
+    );
+  }
+
+  ngOnDestroy() {
+    this.subscription?.unsubscribe();
+    this.updateSub?.unsubscribe();
+    this.deleteSub?.unsubscribe();
+  }
+
+  handleUpdate() {
+    this.profileService.updateUser(this.id, {
+      email: this.email,
+      firstname: this.firstname,
+      lastname: this.lastname,
+      oldPassword: this.oldPassword,
+      newPassword: this.newPassword,
+      newPasswordRepeat: this.newPasswordRepeat,
+    });
+    this.updateSub = this.profileService.updateProfileData.subscribe((data) => {
+      if (data instanceof HttpErrorResponse) {
+        this.snackBar.open(this.t.instant(data?.error?.message), undefined, {
+          panelClass: 'snackbar-error',
+        });
+
+        return;
+      }
+
+      this.id = data.id as string;
+      this.firstname = data.firstname as string;
+      this.lastname = data.lastname as string;
+      this.email = data.email as string;
+      this.firstname = data.firstname as string;
+      this.lastname = data.lastname as string;
+
+      this.snackBar.open(
+        this.t.instant('PROFILES.PROFILE_UPDATE_SUCCESS'),
+        undefined,
+        {
+          panelClass: 'snackbar-success',
+        }
+      );
+    });
+  }
+
+  handleDelete() {
+    const dialogRef = this.dialog.open(ProfileDeleteDialogComponent, {
+      data: { id: this.id },
+    });
+
+    dialogRef.afterClosed().subscribe((result) => {
+      if (result === true) {
+        this.profileService.deleteUser(this.id);
+        this.deleteSub = this.profileService.deleteUserData.subscribe(
+          (data) => {
+            if (data instanceof HttpErrorResponse) {
+              this.snackBar.open(
+                this.t.instant('PROFILES.PROFILE_DELETE_ERROR'),
+                undefined,
+                {
+                  panelClass: 'snackbar-error',
+                }
+              );
+              return;
+            }
+
+            this.profileService.logout();
+
+            this.snackBar.open(
+              this.t.instant('PROFILES.PROFILE_DELETE_SUCCESS'),
+              undefined,
+              {
+                panelClass: 'snackbar-success',
+              }
+            );
+
+            this.router.navigate(['/']);
+          }
+        );
+      }
+    });
+  }
+
+  handleLogout() {
+    this.profileService.logout();
+
+    this.snackBar.open(this.t.instant('PROFILES.LOGOUT_SUCCESS'), undefined, {
+      panelClass: 'snackbar-success',
+    });
+
+    this.router.navigate(['/']);
+  }
+}
+
+@Component({
+  selector: 'ccn-profile-delete-dialog',
+  templateUrl: 'profile-delete-dialog.component.html',
+  styleUrls: ['./profile.component.scss'],
+})
+export class ProfileDeleteDialogComponent {
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public data: DialogData,
+    public dialogRef: MatDialogRef<ProfileDeleteDialogComponent>
+  ) {}
+
+  onNoClick(): void {
+    this.dialogRef.close(false);
+  }
+
+  onYesClick(): void {
+    this.dialogRef.close(true);
+  }
+}
```

Comment:
hast du zwei Komponente in einer implementiert???
wenn ja dann würde ich die 2. auslagern